coordinator/src/orderbook/mod.rs

```diff
@@ -171,27 +376,38 @@ impl Orderbook {
         Ok(message)
     }
 
-    async fn remove_order(
-        &mut self,
-        trader_pubkey: PublicKey,
-        order_id: Uuid,
-    ) -> Result<Option<Message>> {
+    /// Removes a limit order from the orderbook. Ignores removing market orders, since they aren't
+    /// stored into the orderbook.
+    async fn remove_order(&mut self, order_id: Uuid) -> Result<Option<Message>> {
+        // TODO(holzeis): We might want to delay that action in a regular task allowing us to
+        // process orderbook updates much faster in memory and postpone the expensive database
+        // transactions.
         let order = spawn_blocking({
             let mut conn = self.pool.clone().get()?;
             move || {
-                let order = orders::delete_trader_order(&mut conn, order_id, trader_pubkey)?;
+                let matches = matches::get_matches_by_order_id(&mut conn, order_id)?;
+
+                let order = if !matches.is_empty() {
+                    // order has been at least partially matched.
+                    let matched_quantity = matches.iter().map(|m| m.quantity).sum();
+
+                    orders::set_order_state_partially_taken(&mut conn, order_id, matched_quantity)
```

Comment:
❓I don't understand what's going on here. Why do we need to do this when a limit order is being removed?

Reply:
Because it might have been partially matched already.

Today we are setting an order to deleted regardless if it has been matched or not. IMHO we should only set open orders to deleted, but not taken ones.

However how to deal with orders that have been partially matched? I opted to set the state to taken and reduce the quantity to the amount that has already been matched.

Reply:
Shouldn't we do that earlier and not on deletion though. Also, why do we go directly against the database instead of updating the state in the ordebook cache? It seems like the state is only in the DB? Why is that?